src/ensembl/src/content/app/browser/browser-region-field/BrowserRegionField.test.tsx

```diff
@@ -118,79 +109,75 @@ describe('<BrowserRegionField />', () => {
 
-      expect(wrapper.find(Tooltip).props().children).toBe(startError);
+      userEvent.clear(input);
+      userEvent.type(input, 'foo{enter}');
 
-      jest.restoreAllMocks();
+      const errorMessageElement = await screen.findByText(startError);
+      expect(errorMessageElement).toBeTruthy();
+      expect(errorMessageElement.classList.contains('tooltip')).toBe(true);
     });
 
     describe('on validation success', () => {
-      const regionId = faker.lorem.words();
+      const regionId = 'new_chromosome';
 
       beforeEach(() => {
         jest
           .spyOn(browserHelper, 'validateRegion')
           .mockImplementation(
-            async (params: {
-              regionInput: string;
-              genomeId: string | null;
-              onSuccess: (regionId: string) => void;
-              onError: (
-                errorMessages: browserHelper.RegionValidationErrors
-              ) => void;
-            }) => {
-              params.onSuccess(regionId);
-            }
+            async ({
+              onSuccess
+            }: {
+              onSuccess: (arg: string) => void;
+            }): Promise<void> => onSuccess(regionId)
           );
       });
 
-      afterEach(() => {
-        jest.restoreAllMocks();
-      });
-
-      test('changes the focus object if regionInput has a stick/chromosome', () => {
-        const stick = faker.random.number();
-        const start = faker.random.number();
-        const end = faker.random.number();
-        const regionInput = `${stick}:${start}-${end}`;
-
-        wrapper
-          .find(Input)
-          .simulate('change', { target: { value: regionInput } });
-        wrapper.find('form').simulate('submit');
+      it('switches to a different chromosome if it exists in the input', () => {
+        const oldChrLocation = ['13', 1, 1000] as ChrLocation;
+        const newChrLocation = ['12', 1, 1000];
+        const { container } = render(
+          <BrowserRegionField {...defaultProps} chrLocation={oldChrLocation} />
+        );
+        const input = container.querySelector('input') as HTMLInputElement;
 
-        expect(wrapper.props().changeFocusObject).toHaveBeenCalledWith(
-          regionId
+        userEvent.type(
+          input,
+          `${newChrLocation[0]}:${newChrLocation[1]}-${newChrLocation[2]}{enter}`
         );
+
+        expect(defaultProps.changeFocusObject).toHaveBeenCalledWith(regionId);
       });
 
-      test('changes the browser location in same region if regionInput has only start and end (no stick/chromosome)', () => {
-        const start = faker.random.number();
-        const end = faker.random.number();
-        const regionInput = `${start}-${end}`;
+      it('preserves the same chromosome if input contains only new start and end coordinates', () => {
+        const oldChrLocation = ['13', 1, 1000] as ChrLocation;
+        const newChrLocation = ['13', 500, 1000];
```

Comment:
Can't we use the same old and new location arrays used in the above test case? If we can then maybe we don't need to define them separately in individual test cases?

Reply:
These two tests check different things. The previous one checks what happens when the chromosome changes but the location remains the same. This one checks what happens when the location changes but the chromosome stays the same. Don't think it's worth changing.